Sources/SwiftParser/Specifiers.swift

```diff
@@ -827,4 +861,35 @@ extension Parser {
     }
     return RawUnexpectedNodesSyntax(unexpected, arena: self.arena)
   }
+
+  /// Create a version of the same attribute with all tokens missing.
+  class TokenMissingMaker: SyntaxRewriter {
+    override func visit(_ token: TokenSyntax) -> TokenSyntax {
+      return .init(token.tokenKind, presence: .missing)
+    }
+  }
+
+  func withAllTokensMarkedMissing<T: RawSyntaxNodeProtocol>(source: T) -> T {
+    let tokenMissingMaker = TokenMissingMaker(arena: self.arena)
+    let allMissing = tokenMissingMaker.rewrite(
+      Syntax(raw: RawSyntax(source), rawNodeArena: self.arena)
+    ).raw
+    return allMissing.cast(T.self)
+  }
+}
```

Comment:
I tentatively placed the code here, not knowing the ideal source file to house it.

Reply:
How about placing it in Parser.swift, making `TokenMissingMaker` `private` and also using `withAllTokensMarkedMissing` in [Declarations.swift](https://github.com/swiftlang/swift-syntax/blob/70e374101af89aaf998d04587455f0f8d4970751/Sources/SwiftParser/Declarations.swift#L1284-L1293)?